src/views/product-tutorials-view/server.ts

```diff
@@ -0,0 +1,34 @@
+import { getAllCollections } from 'lib/learn-client/api/collection'
+import { getProduct } from 'lib/learn-client/api/product'
+import {
+  Collection as ClientCollection,
+  ProductOption,
+  Product as ClientProduct,
+} from 'lib/learn-client/types'
+import { stripUndefinedProperties } from 'lib/strip-undefined-props'
+import { filterCollections } from './helpers'
+
+export interface ProductTutorialsPageProps {
+  collections: ClientCollection[]
+  product: ClientProduct
+}
+
+export async function getProductTutorialsPageProps(
+  productSlug: ProductOption
+): Promise<{ props: ProductTutorialsPageProps }> {
+  const product = await getProduct(productSlug)
```

Comment:
We store `product` data in the learn database that _sort of_ mirrors the [static json file](https://github.com/hashicorp/dev-portal/blob/main/src/data/waypoint.json) in dev portal now. I'm going to prefer leaning on the learn API where possible to remain consistent with the [product definitions](https://github.com/hashicorp/tutorials/blob/main/content/products/waypoint.yml) that authors can edit.

Reply:
I dig this approach - definitely interesting to speculate if / when the product definitions will somehow converge, but until that day, big +1 to keeping the learn product definitions (and API in general) as the source of truth for learn stuff 👍